### PR TITLE
[BUGFIX] Put events in targetEl if response.json has data

### PR DESCRIPTION
**Pull request by Hasan-Shaker**
Hello,
I had to modify some of the JS to make it work.

### CODE REVIEW

File: Resources/Public/JavaScript/theodia-more.js [hunk 1]
```diff
@@ -32,13 +32,17 @@ document.addEventListener('DOMContentLoaded', function () {
...
+                        targetEl.dataset.events = data.numberEvents;
+                    }
                     if (!data.hasMore) {
                         targetEl.innerHTML = data.html;
                         targetEl.dataset.events = data.numberEvents;
                         // Hide the show more button if not needed anymore
                         element.style.display = 'none';
                     }
-                });
+                }).catch(error => console.error('Error:', error));
```
Comment (xperseguers) [inline Resources/Public/JavaScript/theodia-more.js:45]: I get this line

File: Resources/Public/JavaScript/theodia-more.js [hunk 2]
```diff
@@ -32,13 +32,17 @@ document.addEventListener('DOMContentLoaded', function () {
                 body: formData
             }).then(response => response.json())
                 .then(data => {
+                    if (data.html) {
```
Comment (xperseguers) [inline Resources/Public/JavaScript/theodia-more.js:35]: I can get this check, unsure if really needed, but doesn't hurt.

File: Resources/Public/JavaScript/theodia-more.js [hunk 3]
```diff
@@ -32,13 +32,17 @@ document.addEventListener('DOMContentLoaded', function () {
...
+                        targetEl.innerHTML += data.html;
```
Comment (xperseguers) [inline Resources/Public/JavaScript/theodia-more.js]: I don't get that.

File: Resources/Public/JavaScript/theodia-more.js [hunk 4]
```diff
@@ -32,13 +32,17 @@ document.addEventListener('DOMContentLoaded', function () {
...
+                        targetEl.dataset.events = data.numberEvents;
+                    }
                     if (!data.hasMore) {
```
Comment (xperseguers) [inline Resources/Public/JavaScript/theodia-more.js:39]: To me this check is wrong and the lines

```
targetEl.innerHTML = data.html
targetEl.dataset.events = data.numberEvents;
```

are supposed to go outside of the `if`

but I don't get your patch.

**Comment by Hasan-Shaker** [comment]
OK

**Pull request closed without merging**